app/src/main/java/com/example/courses/presentation/ui/MainFragment.kt

```diff
@@ -259,27 +262,43 @@ class MainFragment : Fragment(), ItemClickListener {
             addedPersonBroadcast,
             IntentFilter(Const.ADD_PERSON_ACTION)
         )
-        sensorManager.registerListener(acselerometorListener,sensor,SensorManager.SENSOR_DELAY_NORMAL)
+        sensorManager.registerListener(
+            acselerometorListener,
+            sensor,
+            SensorManager.SENSOR_DELAY_NORMAL
+        )
 
         if (ActivityCompat.checkSelfPermission(
                 requireContext(),
                 Manifest.permission.ACCESS_FINE_LOCATION
-            ) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(
+            ) != PackageManager.PERMISSION_GRANTED || ActivityCompat.checkSelfPermission(
                 requireContext(),
                 Manifest.permission.ACCESS_COARSE_LOCATION
             ) != PackageManager.PERMISSION_GRANTED
         ) {
-            // TODO: Consider calling
-            //    ActivityCompat#requestPermissions
-            // here to request the missing permissions, and then overriding
-            //   public void onRequestPermissionsResult(int requestCode, String[] permissions,
-            //                                          int[] grantResults)
-            // to handle the case where the user grants the permission. See the documentation
-            // for ActivityCompat#requestPermissions for more details.
-            return
+            ActivityCompat.requestPermissions(
+                requireActivity(),
+                arrayOf(
+                    Manifest.permission.ACCESS_FINE_LOCATION,
+                    Manifest.permission.ACCESS_COARSE_LOCATION
+                ),
+                REQUEST_CODE_PERMISSION_LOCATION
+            )
+
```

Comment:
Assume, you could have a crash here, if deny permission. 
Add 'return' statement here. 
Also, you have to add 'onRequestPermissionsResult' method where you could start location updates in case of success.

Reply:
Actually, I've got a crash on line 289 even before I'm able to grant or deny the permission.